rustls/src/crypto/mod.rs

```diff
@@ -226,28 +222,15 @@ impl CryptoProvider {
     /// Call this early in your process to configure which provider is used for
     /// the provider.  The configuration should happen before any use of
     /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
-    #[cfg(feature = "std")]
     pub fn install_default(self) -> Result<(), Arc<Self>> {
-        PROCESS_DEFAULT_PROVIDER.set(Arc::new(self))
-    }
-
-    /// Sets this `CryptoProvider` as the default for this process.
-    ///
-    /// This can be called successfully at most once in any process execution.
-    ///
-    /// Call this early in your process to configure which provider is used for
-    /// the provider.  The configuration should happen before any use of
-    /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
-    #[cfg(not(feature = "std"))]
-    pub fn install_default(self) -> Result<(), Box<Arc<Self>>> {
-        PROCESS_DEFAULT_PROVIDER.set(Box::new(Arc::new(self)))
+        static_default::install_default(Arc::new(self))
```

Comment:
Suggest you change `static_default::install_default()` to just take `CryptoProvider`, and let it handle the `Arc`-wrapping.

Reply:
Done ... my impression is that this could add another move operation on the `CryptoProvider` config data (move, not copy), but this is just my impression ... suspect that any possible performance impact should be negligible, if any. Definitely seems to be cleaner this way :)